### PR TITLE
Clone overwrite

### PR DESCRIPTION
**Pull request by scottnath**
@e2tha-e 
cc: @rebmullin @sergesemashko @conortm @gvorbeck @beynya @nikkiana

This PR is for [TA99424](https://rally1.rallydev.com/#/26707410139ud/detail/task/41508573767) - local overrides.

This allows the addition of a flag on `gulp clone` commands to allow overwriting the local version of a pattern - if it exists.

An overwrite clone statement would be like so:

```
gulp clone --pattern node_modules/pattern-library/patterns/atoms/text/h5 --overwrite YES
```

I've also made some slight adjustments to logger.js because I was running into errors when trying to use the system. Note: cannot get exitOnError to work - but otherwise logger works as expected.

thanks,
Scott

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -4,27 +4,42 @@
...
+  var projectOptions = getOptions.getProjectOptions();
 
-  var transports = [
+  // merge project options with default options; project options take precedence
+  var options = getOptions.mergeOptions(defaultOptions, projectOptions);
+
+  // check if we're unit testing
+  if (isTest()) {
+    options.logger.silent = true;
+  }
```
Comment (e2tha-e) [inline lib/logger.js]: @scottnath This refactor will suppress errors and warnings in tests. I don't think this is what we want.

Comment (scottnath) [inline lib/logger.js]: @e2tha-e 

i see - could it instead be something like

```if (isTest() && (some check if this is a warning/test) {```

?

File: lib/logger.js [hunk 2]
```diff
@@ -4,27 +4,42 @@
...
+  // get the project's options
+  var projectOptions = getOptions.getProjectOptions();
 
-  var transports = [
+  // merge project options with default options; project options take precedence
+  var options = getOptions.mergeOptions(defaultOptions, projectOptions);
+
+  // check if we're unit testing
+  if (isTest()) {
+    options.logger.level = 'warn';
```
Comment (scottnath) [inline lib/logger.js]: @e2tha-e properly limiting for errors during tests now

**Comment by e2tha-e** [comment]
@scottnath Looks good :+1: Merging

**Comment by scottnath** [comment]
Thank you @e2tha-e

**Pull request merged**